src/cacheHttpClient.ts

```diff
@@ -118,17 +128,19 @@ export async function downloadCache(
 
 // Reserve Cache
 export async function reserveCache(key: string): Promise<number> {
-    const restClient = createRestClient();
+    const httpClient = createHttpClient();
 
     const reserveCacheRequest: ReserveCacheRequest = {
         key
     };
-    const response = await restClient.create<ReserveCacheResponse>(
-        "caches",
+    const additionalHeaders = {
+        "Content-Type": "application/json"
```

Comment:
I think we should push a change to the http-client so that `postJson` and `getJson` have these headers automatically.

Reply:
Id like to, but Im not sure how much work it would be (probably not much). Setting the headers for that is easy but we would also need to have code to merge existing headers, making sure not to overwrite any existing content-type header. Ill at least investigate, it shouldnt be a lot of code + a few tests

Reply:
merge headers exists: https://github.com/actions/http-client/blob/master/index.ts#L405

Reply:
So, let's make the change.  I agree with @joshmgross - it's the reason we have helper methods like these.  Should be a 5 min change.  I'm available to release